src/graph.rs

```diff
@@ -1616,6 +1643,54 @@ fn is_untyped(media_type: &MediaType) -> bool {
   )
 }
 
+/// Check import assertions.
+fn check_import_assertions(
+  import: &Import,
+  maybe_module_metadata: Option<&ModuleMetadata>,
+) -> Vec<ImportError> {
+  // Skip assertion checks for dynamic imports for now.
```

Comment:
Is this done later? Why aren't dynamic imports checked? Maybe include the answer to "why?" in the comment here.

Reply:
The way we discover dynamic module slot errors in `deno_graph` and defer elevating them until runtime, there's no good mechanism for doing that currently in CLI for resolution/import errors so we just leave it for `deno_core`. That and there was a test for it (`test_build_graph_dynamic_json_ignores_assert`) so I just excluded it for simplicity.

I think I'll move this check to `validate()` actually, it's more consistent with module slot errors.

Reply:
Good idea. That sounds a lot better.